### ISSUE TITLE
chromium-gost и Госуслуги

### ISSUE THREAD

**Issue by dymn** [msva/mva-overlay#187]
При попытке открыть https://www.gosuslugi.ru или авторизоваться с ЕИС https://eruz.zakupki.gov.ru/auth/login выдают ошибку:
Не удается получить доступ к сайту. Веб-страница по адресу https://eruz.zakupki.gov.ru/auth/login, возможно, временно недоступна или постоянно перемещена по новому адресу.
ERR_FAILED
В консоли следующие ошибки:
```
[10258:10280:0820/190133.736968:ERROR:nss_util.cc(345)] After loading Root Certs, loaded==false: NSS error code: -8018
[10315:10323:0820/190133.751935:ERROR:openssl_ssl_util.cc(181)] OpenSSL SYSCALL error, earliest error code in error queue: 0: Неизвестная ошибка -1 (-1)
[10315:10323:0820/190133.752005:ERROR:ssl_client_socket_impl.cc(1027)] handshake failed; returned 0, SSL error code 5, net_error -2
[10315:10323:0820/190134.892158:ERROR:openssl_ssl_util.cc(181)] OpenSSL SYSCALL error, earliest error code in error queue: 0: Выполнено (0)
[10315:10323:0820/190134.892171:ERROR:ssl_client_socket_impl.cc(1027)] handshake failed; returned 0, SSL error code 5, net_error -2
[10315:10323:0820/190135.274962:ERROR:openssl_ssl_util.cc(181)] OpenSSL SYSCALL error, earliest error code in error queue: 0: Неизвестная ошибка -1 (-1)
```
Пробовал переделывать ebuild под последнюю версию chromium-gost-127.0.6533.99 - результат такой же.
Устанавливал dev-libs/openssl-3.3.1-r3 вместо стабильного 3.0.13-r2 - без результата.
Что-то с доступом к установленным сертификатам?
На Госуслуги все норм заходит через установленный тут же ff-129 и Chrome-127.0.6533.119.
Системных прокси не стоит.

**Comment by msva** [comment]
Да, у меня тоже не получилось добиться его работы с ГОСТ-шифрованием.

Вообще, формально, он, вроде как, должен испольховать Крипто-ПРО для ГОСТ (как и Яндекс-Браузер), но никто из них у меня не смог открыть ни один сайт с форсированным ГОСТ-шифрованием.

А конкретно у chromium-gost, там ещё, кажется и настройка режимов работы с этими алгоритамами, кажется, по умолчанию так выставлена, что если сайт хотя бы предлагает (а не обязывает даже), то ГОСТ будет в приоритете.
И у меня он тоже не открывал много чего.
Я как-то экспериментировал с этим, но так и не хватило времени добить до полностью рабочего состояния (и к текущему моменту я даже подзабыл даже где именно те параметры касательно режимов можно подкручивать).

Попробую разобраться с этим когда закончится очередной авральный период по работе...

**Comment by dymn** [comment]
Может что-то как раз с Крипто-ПРО не так?
Я ставил с вашего ебилда его и была проблема:
при попытке установить сертификаты из контейнера в консоли была ошибка "Библиотека поставщика проинициализирована неправильно" и вылетало окно "csp license is not set or corrupted". Немного погуглил, добавил демолицензию:
/opt/cprocsp/sbin/amd64/cpconfig -license -set 5050U-C0037-EKP59-NAXWV-WMCWE

После этого установились сертификаты из контейнера.

app-crypt/ifcplugin-bin-3.1.1.0 пришлось ставить из guru, но там ошибка в контрольной сумме - скопировал ебилд в локальный реп и пересоздал манифест. Да и странный он - архив из red-soft.ru, а не из самого cryptopro. Но я писать ebuild так и не научился, приходится использовать уже имеющееся.

Уж очень достало загружать Virtualbox то с виндой, то с Linux mint, где все работает как должно.

**Comment by msva** [comment]
По поводу лицензии - вообще, при установке (по крайней мере 12900 версии) демо-лицензия должна активировать, это прописано в postinstall-скрипте. Так что странно.
Посмотрю этот момент тоже.

По поводу ifcplugin - он есть в моём оверлее под названием "gosuslugi-plugin" (впрочем, учитывая, что у не-версионированная ссылка на архив, он иногда тоже может протухать)

**Comment by msva** [comment]
Я удалил chromium-gost из оверлея, т.к. так и не получилось заставить его работать с gost-шифрованием на сайтах, которые без него не работают (типа lkulgost.nalog.ru), а без этого от него нет особого смысла.

Однако, у меня получилось заставить с гостовым шифрованием работать яндекс-браузер (в оверлее yandex).
Советую использовать его.

Так же, для госуслуг в этом оверлее есть плагин gosplugin. Который, в целом, тоже проверен и работает, хотя есть некоторые визуальные недочёты (по крайней мере у меня, при тёмной теме в системе, на форме, которую выводит плагин для ввода пин-кода от токена, не рендерятся надписи), но в целом работать можно.

Если получится - когда-нибудь придумаю и как пофиксить это.

Так же, когда представится возможность (в последнее время очень много работы и не хватает выходных чтобы разобраться с тем, что накапливается за неделю) - постараюсь выпустить и новую версию cprocsp (текущая, впрочем, до сих пор работает), и переработать ебилд к более правильному виду.
